src/quota_aggregator_impl.cc

```diff
@@ -57,11 +57,15 @@ QuotaAggregatorImpl::CacheElem::ReturnAllocateQuotaRequestAndClear(
     *(request.mutable_allocate_operation()) =
         operation_aggregator_->ToOperationProto();
     operation_aggregator_ = NULL;
+    request.mutable_allocate_operation()->set_quota_mode(
+        QuotaOperation::BEST_EFFORT);
   } else {
     // If requests are not aggregated, use the stored initial request
     // to allocate minimum token
     request = quota_request_;
-  }
+    request.mutable_allocate_operation()->set_quota_mode(
```

Comment:
Why are we modifying the quota request here? If aggregation is turned off, then the caller is responsible for sending the correct request, quota cache should be NOOP.

Reply:
By "caller" i mean the client using the quota cache